File: day06/universal_orbit_map.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.30);
use experimental qw(signatures);
use Graph::Undirected;
use List::Util qw(sum);

my $g = Graph::Undirected->new;
while (<>) {
    chomp;
    my @a = split /\)/;
    $g->add_edge(split /\)/);
}

my $sum = 0;
for my $v ($g->vertices) {
    my @a = $g->SP_Dijkstra("COM", $v);
    $sum += @a - 1;
}
say "Part 1: $sum";

my @path = $g->SP_Dijkstra("YOU", "SAN");
say "Part 2: ", @path - 3;

File: day06/universal_orbit_map.py
#!/usr/bin/env python3
from sys import argv
import networkx as nx

G = nx.Graph()
filename = argv[1]
with open(filename) as f:
    for line in f:
        G.add_edge(*line.rstrip().split(')'))
print('Part 1:', sum(nx.algorithms.shortest_path_length(G, 'COM').values()))
print('Part 2:', len(nx.algorithms.shortest_path(G, 'YOU', 'SAN'))-3)
